flit_core/flit_core/config.py

```diff
@@ -142,6 +142,27 @@ def prep_toml_config(d, path):
             dtool['sdist'].get('exclude', []), 'exclude'
         )
 
+    data_dir = dtool.get('external-data', {}).get('directory', None)
+    if data_dir is not None:
+        toml_key = "tool.flit.external-data.directory"
+        if not isinstance(data_dir, str):
+            raise ConfigError(f"{toml_key} must be a string")
```

Comment:
Do you want to add tests for these error branches, that codecov is complaining aren't covered?

Reply:
It feels too much like testing that the code I've written is the code that's there, which always feels a bit pointless. If someone wants to add some test cases and push coverage up a bit, I'm happy enough to merge it, but I don't feel like it's worth spending time on.

If someone finds a case where it's currently doing the wrong thing, I'm happier to add a test for that as we fix it.